### PR TITLE
fix equality mapper functions

### PR DESCRIPTION
**Pull request by matthiasdiener**

### CODE REVIEW

**Review by kaushikcfd — APPROVED** [review]
LGTM, thanks!

File: pytato/equality.py [hunk 1]
```diff
@@ -230,7 +231,8 @@ def map_loopy_call(self, expr1: LoopyCall, expr2: Any) -> bool:
                 )
 
     def map_loopy_call_result(self, expr1: LoopyCallResult, expr2: Any) -> bool:
-        return (self.rec(expr1._container, expr2._container)
+        return (expr1.__class__ is expr2.__class__
+                and self.rec(expr1._container, expr2._container)
                 and expr1.tags == expr2.tags)
```
Comment (matthiasdiener) [inline pytato/equality.py:236]: Should we also compare the other fields here (name, axes)?

Comment (kaushikcfd) [reply pytato/equality.py:236]: Yep!

Comment (matthiasdiener) [reply pytato/equality.py]: Done in f72bb2d

**Pull request merged**